src/context/Authentication/AuthState.js

```diff
@@ -56,14 +71,28 @@ const AuthState = ({ children }) => {
       if (res.status === 200) {
         console.log(res.data);
         dispatch({
-          type: 'SET_USER',
+          type: "SET_USER",
           payload: res.data,
         });
       }
     });
   };
```

Comment:
The effect will be run each time the component renders because you have omitted the dependencies. If you see, the effect depends on "loadPreviousThemePreferences() , getToken(), setLoading() and getUser()". So it would be better to call the effect only when the component renders for the first time and then for the stuff inside the effect. To do this, we can call the effect with an empty array of dependencies. Now the effect will be only called for the initial render and also, for the stuff (inside the effect) changes. I think it would be efficient.
**Note** The functions inside the effect should be defined in the component's scope (not imported) otherwise the effect will be called only once even if the function calls over and over again. It is the rule of useEffect.

Reply:
sure